src/main/java/kafka/vertx/demo/PeriodicProducer.java

```diff
@@ -54,7 +55,7 @@ public void start(Promise<Void> startPromise) {
       });
   }
 
-  private void setup(JsonObject config, Promise<Void> startPromise) {
+  private Future<Void> setup(JsonObject config) {
```

Comment:
If this is syncronous, return void. Then change compose() above to map()

Reply:
I've pushed up a new version